sdk/core/azure-core/src/main/java/com/azure/core/util/polling/LocationPollingStrategy.java

```diff
@@ -0,0 +1,179 @@
+    private static final String HTTP_METHOD = "httpMethod";
+    private static final String RETRY_AFTER = "Retry-After";
+    private static final String POLL_RESPONSE_BODY = "pollResponseBody";
+
+    private final JacksonAdapter serializer = new JacksonAdapter();
+    private final ClientLogger logger = new ClientLogger(LocationPollingStrategy.class);
+
+    private final HttpPipeline httpPipeline;
+    private final Context context;
+
+    /**
+     * Creates an instance of the location polling strategy.
+     *
+     * @param httpPipeline an instance of {@link HttpPipeline} to send requests with
+     * @param context additional metadata to pass along with the request
+     */
+    public LocationPollingStrategy(
+            HttpPipeline httpPipeline,
+            Context context) {
+        this.httpPipeline = httpPipeline;
+        this.context = context;
+    }
+
+    @Override
+    public Mono<Boolean> canPoll(Response<?> initialResponse) {
+        HttpHeader locationHeader = initialResponse.getHeaders().get(LOCATION);
+        return Mono.just(locationHeader != null);
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public Mono<LongRunningOperationStatus> onInitialResponse(Response<?> response, PollingContext<T> pollingContext,
+                                                   TypeReference<T> pollResponseType) {
+        HttpHeader locationHeader = response.getHeaders().get(LOCATION);
+        if (locationHeader != null) {
+            pollingContext.setData(LOCATION, locationHeader.getValue());
+        }
+        pollingContext.setData(HTTP_METHOD, response.getRequest().getHttpMethod().name());
+        pollingContext.setData(REQUEST_URL, response.getRequest().getUrl().toString());
+
+        if (response.getStatusCode() == 200
+                || response.getStatusCode() == 201
+                || response.getStatusCode() == 202
+                || response.getStatusCode() == 204) {
+            return Mono.just(LongRunningOperationStatus.IN_PROGRESS);
+        } else {
+            throw logger.logExceptionAsError(
+                new RuntimeException("Operation failed or cancelled: " + response.getStatusCode()));
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public Mono<PollResponse<T>> poll(PollingContext<T> pollingContext, TypeReference<T> pollResponseType) {
+        HttpRequest request = new HttpRequest(HttpMethod.GET, pollingContext.getData(LOCATION));
+        Mono<HttpResponse> responseMono;
+        if (context == null) {
+            responseMono = httpPipeline.send(request);
+        } else {
+            responseMono = httpPipeline.send(request, context);
+        }
+        return responseMono.flatMap(res -> {
+            HttpHeader locationHeader = res.getHeaders().get(LOCATION);
+            if (locationHeader != null) {
+                pollingContext.setData(LOCATION, locationHeader.getValue());
+            }
```

Comment:
Will it be different from initial response?

Reply:
I'm not sure - but if a new location header is received, I think it's best to honor the new one.